api/services/contact_fields_mapping_service.py

```diff
@@ -0,0 +1,119 @@
+from api import models, db, helpers
+
+
+class ContactFieldsMappingService(object):
+    def __init__(self):
+        self.user_phone = None
+
+    def set_init_data(self, jsonData):
+        user_phone = helpers.fetch_by_key("urn", jsonData["contact"])
+        self.user_phone = helpers.sanitize_phone_string(user_phone)
+
+    def handle_contact_fields_data(self, jsonData):
+        try:
+            self.set_init_data(jsonData)
+            user_details = models.User.query.get_by_phone(self.user_phone)
+            if not user_details:
+                return False
+
+            custom_fields = jsonData["contact"]["fields"]
+
+            if custom_fields:
+                for key in custom_fields:
+                    field_value = custom_fields[key]
+                    if key in custom_fields and field_value:
```

Comment:
This is also an unnecessary check as key will always be there because we are iterating based on it. 
So we just need to check the value
```suggestion
                    if field_value:
```

Reply:
Okay, got it.